src/sqlfluff/core/rules/std/L020.py

```diff
@@ -78,58 +94,60 @@ def _get_aliases_from_select(cls, segment, dialect=None):
                 value_table_function_aliases.append(alias_info)
         return table_aliases, value_table_function_aliases
 
-    def _eval(self, segment, parent_stack, **kwargs):
-        """Get References and Aliases and allow linting.
-
-        This rule covers a lot of potential cases of odd usages of
-        references, see the code for each of the potential cases.
-
-        Subclasses of this rule should override the
-        `_lint_references_and_aliases` method.
-        """
-        if segment.is_type("select_statement"):
-            table_aliases, value_table_function_aliases = self._get_aliases_from_select(
-                segment, kwargs.get("dialect")
-            )
-            if not table_aliases and not value_table_function_aliases:
-                return None
-
-            # Iterate through all the references, both in the select clause, but also
-            # potential others.
-            sc = segment.get_child("select_clause")
-            reference_buffer = list(sc.recursive_crawl("object_reference"))
-            # Add any wildcard references
-            reference_buffer += list(sc.recursive_crawl("wildcard_identifier"))
-            for potential_clause in (
-                "where_clause",
-                "groupby_clause",
-                "having_clause",
-                "orderby_clause",
+    @classmethod
+    def get_select_statement_info(
```

Comment:
It doesn't feel right to me that you'd use a Rule method as a "util" or a "library", the only reason I would see for a Rule to use another rule is similarity in what it's fixing/looking for (through inheritance)

Do you think we could move this method and the _get_aliases_from_select into a `SelectCrawler` maybe? Which would inherit from `BaseCrawler` ?

Reply:
happy to be convinced otherwise haha

Reply:
I agree completely! I was hoping to get feedback from some folks before doing this, for suggestions on what directory and/or module to put it in.

@NiallRees, @alanmcruickshank: Any thoughts on the above? Happy to follow advice or forge my own path here. 🙂

Reply:
@dmateusp and others: I'm 💯 on the suggested reorg, but if it's okay, I suggest maybe deferring that until a future PR. This could be an immediate follow-on PR, or perhaps wait until the next rule that uses this code.

My reason for suggesting this is that it's challenging to keep a "new rule" PR current. In the week or so this PR has been open, I think I've already had to renumber the rules twice. (Each time a new rule is added, I get a merge conflict and have to make a number of small code changes related to the renumbering, just to get the tests and everything working again.)

Reply:
Since this PR has been open a bit, I will look into the reorg. I think I will look at creating a new module with functions, rather than a class deriving from `BaseCrawler`. My reasoning is, I think over time we may start to assemble a growing number of code analysis tools, and being able to (re-)use them without inheritance will be more flexible (like if one rule wants to use multiple tools).